pkg/system/chtimes_nowindows.go

```diff
@@ -13,3 +16,17 @@ import (
 func setCTime(path string, ctime time.Time) error {
 	return nil
 }
+
+// setAMTimeNoFollow will set access/modification time on a file,
+// without following symbol link.
+func setAMTimeNoFollow(path string, atime time.Time, mtime time.Time) error {
+	uts := []unix.Timespec{
+		unix.NsecToTimespec(atime.UnixNano()),
+		unix.NsecToTimespec(mtime.UnixNano()),
+	}
+	err := unix.UtimesNanoAt(unix.AT_FDCWD, path, uts, unix.AT_SYMLINK_NOFOLLOW)
+	if err == unix.ENOSYS {
+		err = os.Chtimes(path, atime, mtime)
```

Comment:
Under what conditions does this error occur?

I'm  also wondering if we should actually fall back here (because the function is explicitly to "not follow"), and this would "break that promise", so if we should leave that to the caller to decide on falling back to the other approach 🤔

Reply:
Good point~   the behavior should not contradict with its "spec"; 
As for `unix.ENOSYS`, I have no idea on what kind/version of linux system `UtimesNanoAt` would yield that error code ( I just copied it from some other source code...)
`docker save`  prefers "no follow", and definitely should not fail when 'no follow' is not supported.  Maybe I can move the `prefer` logic to `docker save` module, and let me think about it
Thx~

Reply:
Thanks for giving it some thought (apologies for my sometimes "partial" reviews; I try write down thoughts I have when glancing over code. They're sometimes "quick blurbs", so definitely don't hesitate to speak up if they're "wrong" or don't make sense! 😅)

Reply:
Code changed, 
1. ChtimesNoFollow would not resort to default behavior of Chtimes and return error when not supported
2.  `docker save` would first try ChtimesNoFollow, and would resort to Chtimes when "no follow" not supported